activerecord/test/cases/persistence_test.rb

```diff
@@ -117,6 +117,15 @@ def test_increment_attribute
     assert_equal 53, accounts(:signals37, :reload).credit_limit
   end
 
+  def test_increment_aliased_attribute
```

Comment:
@prathamesh-sonpatki I added a test to `counter_cache`, but decided to keep this one here. Let me know if this is a problem.